src/cmd/services/m3dbnode/config/limits.go

```diff
@@ -39,4 +41,15 @@ type Limits struct {
 	// process would pause until some of the repaired bytes had been persisted to disk (and subsequently
 	// evicted from memory) at which point it would resume.
 	MaxOutstandingRepairedBytes int64 `yaml:"maxOutstandingRepairedBytes" validate:"min=0"`
+
+	// MaxRecentlyQueriedBlocks controls the max blocks being queried within a given
+	// lookback period. Queries which are issued while this max is surpassed are abandonded.
+	MaxRecentlyQueriedBlocks *MaxRecentlyQueriedBlocks `yaml:"maxRecentlyQueriedBlocks"`
+}
+
+// MaxRecentlyQueriedBlocks controls the max blocks being queried within a given
+// lookback period. Queries which are issued while this max is surpassed are abandonded.
+type MaxRecentlyQueriedBlocks struct {
```

Comment:
nit: actually, this is a config right? Maybe worth renaming this to `MaxRecentlyQueriedBlocksConfiguration` and `Limits` to `LimitsConfiguration` for consistency?

Reply:
Agreed, as per other types in this package (e.g. cache.go and bootstrap.go) "Limits" should be "LimitsConfiguration" and "MaxRecentlyQueriedBlocks" should be "MaxRecentlyQueriedBlocksConfiguration".